sensors/Vision360.py

```diff
@@ -3,23 +3,107 @@
 from sensors.Sensor import Sensor
 from world.ObjectType import *
 from world.WorldObject import *
+from omnicam.protocol import Mode, COMMUNICATION_PORT, REMOTE_ADDR, MODE_IDENTIFIER_INDICATOR
+from websockets.sync.client import connect as sync_connect
+import websockets
+import threading
+import json
 
+import time
+
+class Vision360SocketError(Exception):
+    def __init__(self, additional_info, e=None, *args):
+        super().__init__(args)
+        self.e = e
+        self.additional_info = additional_info
+    def __str__(self):
+        return(f"An error occured in the 360 vision socket connection (connection to {REMOTE_ADDR}).\n{self.additional_info}\nFull error:{self.e}")
+
+class Vision360Error(Exception):
+    def __init__(self, additional_info, *args):
+        super().__init__(args)
+        self.additional_info = additional_info
+    def __str__(self):
+        return(f"An error occured in the 360 vision system.\n{self.additional_info}")
+
+BRAIN_TO_MODE_MAPPING = {
```

Comment:
This should be a member of the Vision360 class.

Reply:
I was leaving this at module level in case it needed to be shared by other classes within the module.

Reply:
I think it can still be referenced statically if it's inside the class if need be. Everything at the moment because of each class being self contained hasn't needed to do something like that so far however.

In this instance another option would be to put `BRAIN_TO_MODE_MAPPING` inside of `RobotBrain` and then pull it out of the the `RobotBrain` that's passed to the vision system. That's what I've done previously when that sort of information was needed.

Reply:
I'll move it into the class, but I really think it's a non-issue since we have module-level scoping.

Reply:
Just blame it on my strong desire for the patterns to match. :)